packages/astro/src/runtime.ts

```diff
@@ -286,7 +286,7 @@ async function createSnowpack(astroConfig: AstroConfig, options: CreateSnowpackO
   const mountOptions = {
     ...(existsSync(astroConfig.public) ? { [fileURLToPath(astroConfig.public)]: { url: '/', static: true, resolve: false } } : {}),
     [fileURLToPath(frontendPath)]: '/_astro_frontend',
-    [fileURLToPath(src)]: '/_astro/src', // must be last (greediest)
+    [fileURLToPath(src)]: `/_astro/${fileURLToPath(src).slice(fileURLToPath(projectRoot).length - 1)}`, // must be last (greediest)
```

Comment:
This will probably break on Windows, since you'll have `/_astro/foo\bar\baz`

Instead of converting to filepath, this is actually where file URLs really shine! You should be able to do: ``/_astro/${src.pathname.slice(projectRoot.pathname.length - 1)}`, or something like that. If you can test this and get it working, then this LGTM to merge!

Reply:
Thanks for catching this! I changed it and added a 2nd line to allow for a custom `pages/` dir as well. Now I'd like to write some tests for this as well, but I'm not entirely sure what the most efficient way to do it would be. Do you have any idea for how to test if the server uses the correct paths?